alert/alert.py

```diff
@@ -134,8 +138,23 @@ def process_file(filename):
                 series[measure_date] = numpy.array(measure["values"])
             buckets = measure['buckets']
 
-        measure_name = os.path.splitext(os.path.basename(filename))[0] # Filename without extension
-        regressions += compare_histogram(series, measure_name, buckets)
+        measure_name, _ = os.path.splitext(os.path.basename(filename)) # the measure name is the filename without the extension
+
+    if series:
+        # check that the series is valid
+        reference_date, expected_bucket_count = None, None
+        for date, histogram in series.items():
+            if expected_bucket_count is None:
+                reference_date, expected_bucket_count = date, len(histogram)
+            elif len(histogram) != expected_bucket_count:
+                logging.warn(
+                    "BUCKET COUNT MISMATCH - IGNORING HISTOGRAM {} ({} has {} buckets, while {} has {} buckets)".format(
+                        measure_name, reference_date, expected_bucket_count, date, len(histogram)
+                    )
+                )
+                return []
+
+    regressions += compare_histogram(series, measure_name, buckets)
```

Comment:
Conceptually I see it as the code above it generating the value of `series`, and then this code uses it. It feels a bit cleaner, but if necessary I can move it inside the context manager.